### PR TITLE
[terra-dev-site-v7] Migrate secondary nav layout

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

The secondary navigation layout is going to be scoped out of the terra-application v2 release. This PR migrates the layout into terra-dev-site and applies a lowlight theme of my own design. (Suggestions welcome).

Closes #

### Deployment Link

https://terra-dev-site-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "6.30.0",
   "description": "Dynamically builds a react-hash-routed site based on site configuration, navigation configuration and component configuration.",
   "engines": {
-    "node": ">=10"
+    "node": ">=10 <16"
```
Comment (mjhenkes) [inline package.json:6]: building this project with node 16 fails in heroku

File: src/layouts/secondary-navigation-layout/clinical-lowlight-theme/SecondaryNavigationLayout.module.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .clinical-lowlight-theme {
+    --terra-dev-site-secondary-navigation-layout-side-nav-sidebar-background-color: #222a2e;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/clinical-lowlight-theme/SecondaryNavigationLayout.module.scss:5]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/orion-fusion-theme/SecondaryNavigationLayout.module.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .orion-fusion-theme {
+    --terra-dev-site-secondary-navigation-layout-side-nav-sidebar-background-color: #eff0ef;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/orion-fusion-theme/SecondaryNavigationLayout.module.scss:5]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/side-nav/clinical-lowlight-theme/CollapsingNavigationMenu.module.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+:local {
+  .clinical-lowlight-theme {
+    --terra-dev-site-collapsing-navigation-menu-item-background-color: #1c2326;
+    --terra-dev-site-collapsing-navigation-menu-item-hover-background-color: #1c2326;
+    --terra-dev-site-collapsing-navigation-menu-item-focus-outline: #fff;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/side-nav/clinical-lowlight-theme/CollapsingNavigationMenu.module.scss:7]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/side-nav/clinical-lowlight-theme/SideNavHeader.module.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .clinical-lowlight-theme {
+    --terra-dev-site-side-nav-header-page-header-container-box-shadow: inset -3px -4px 0px -3px #1c1f21, inset -3px -5px 0px -3px #C7CACB;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/side-nav/clinical-lowlight-theme/SideNavHeader.module.scss:5]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/side-nav/orion-fusion-theme/CollapsingNavigationMenu.module.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+:local {
+  .orion-fusion-theme {
+    --terra-dev-site-collapsing-navigation-menu-item-background-color: #e1f3fd;
+    --terra-dev-site-collapsing-navigation-menu-item-hover-background-color: #e1f3fd;
+    --terra-dev-site-collapsing-navigation-menu-item-focus-outline: #000;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/side-nav/orion-fusion-theme/CollapsingNavigationMenu.module.scss:7]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/side-nav/orion-fusion-theme/SideNavHeader.module.scss [hunk 7]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .orion-fusion-theme {
+    --terra-dev-site-side-nav-header-page-header-container-box-shadow: inset -3px -4px 0px -3px #fefffe, inset -3px -5px 0px -3px #C7CACB;
+  }
+}
```
Comment (mjhenkes) [inline src/layouts/secondary-navigation-layout/side-nav/orion-fusion-theme/SideNavHeader.module.scss:5]:
```suggestion
}

```

File: src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.jsx [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+            }
+          }}
+        />
+      )}
+      <div
+        className={cx('side-nav-container', theme.className)}
+        ref={pageContainerRef}
+      >
+        <div
+          ref={resizeOverlayRef}
```
Comment (tbiethman) [inline src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.jsx:322]: This guy isn't necessary anymore with the removal of workspace (I think?)

File: src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.module.scss [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+  .side-nav-resize-overlay {
+    bottom: 0;
+    cursor: col-resize;
+    display: none;
+    left: 0;
+    position: absolute;
+    right: 0;
+    top: 0;
+    z-index: 5; // TODO 5 is very important for safari, validate final value for this
```
Comment (tbiethman) [inline src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.module.scss:94]: You could remove this class as well (and not have to worry about the TODO, hurray)

File: src/layouts/secondary-navigation-layout/side-nav/SideNavHeader.jsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  const theme = React.useContext(ThemeContext);
+  return (
+    <div className={cx('page-header-container', theme.className)}>
+      <div className={cx('page-layout-header')}>
+        {onRequestClose ? (
+          <div className={cx('back-button-container')}>
+            <Button
+              className={cx(['header-button', 'back-button'])}
+              icon={<IconLeft />}
+              text="Back" // TODO validate icon/text and intl
```
Comment (tbiethman) [inline src/layouts/secondary-navigation-layout/side-nav/SideNavHeader.jsx:36]: We could lift one of the ActionHeader translations here (Back or Close I dunno which)

**Review by tbiethman — APPROVED** [review]

File: src/layouts/secondary-navigation-layout/side-nav/SideNavHeader.jsx (hunk 10 above)
Comment (mjhenkes) [reply src/layouts/secondary-navigation-layout/side-nav/SideNavHeader.jsx]: I went with back.

**Pull request closed without merging**